server/src/main/java/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -1186,6 +1227,431 @@ protected void pushToUpdateQueue(final List<NetworkVO> networks) throws Interrup
+        final SetMonitorServiceCommand command = new SetMonitorServiceCommand(services);
+        command.setAccessDetail(NetworkElementCommand.ROUTER_IP, getRouterControlIP(router));
+        command.setAccessDetail(NetworkElementCommand.ROUTER_NAME, router.getInstanceName());
+        command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_ENABLED, RouterHealthChecksEnabled.valueIn(router.getDataCenterId()).toString());
+        command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_BASIC_INTERVAL, RouterHealthChecksBasicInterval.value().toString());
+        command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_ADVANCED_INTERVAL, RouterHealthChecksAdvancedInterval.value().toString());
+        command.setAccessDetail(SetMonitorServiceCommand.ROUTER_HEALTH_CHECKS_EXCLUDED, RouterHealthChecksToExclude.valueIn(router.getDataCenterId()));
+        command.setAdditionalData(getAdditionalDataForRouterHealthChecks(router));
+        command.setReconfigureAfterUpdate(reconfigure);
+        command.setDeleteFromProcessedCache(deleteFromProcessedCache); // As part of updating
+        return command;
+    }
+
+    private boolean updateRouterHealthCheckData(DomainRouterVO router) {
+        if (!RouterHealthChecksEnabled.valueIn(router.getDataCenterId())) {
+            return false;
+        }
+
+        SetMonitorServiceCommand command = createMonitorServiceCommand(router, null,true, true);
+        String controlIP = getRouterControlIP(router);
+        if (StringUtils.isBlank(controlIP) || controlIP.equals("0.0.0.0")) {
+            s_logger.warn("Skipping update data on router " + router.getUuid() + " because controlIp is not correct.");
+            return false;
+        }
+
+        s_logger.info("Updating data for router health checks for router " + router.getUuid());
+        Answer origAnswer = null;
+        try {
+            origAnswer = _agentMgr.easySend(router.getHostId(), command);
+        } catch (final Exception e) {
+            s_logger.error("Error while sending update data for health check to router: " + router.getInstanceName(), e);
+            return false;
+        }
+
+        if (origAnswer == null) {
+            s_logger.error("Unable to update health checks data to router " + router.getHostName());
+            return false;
+        }
+
+        GroupAnswer answer = null;
+        if (origAnswer instanceof GroupAnswer) {
+            answer = (GroupAnswer) origAnswer;
+        } else {
+            s_logger.error("Unable to update health checks data to router " + router.getHostName() + " Received answer " + origAnswer.getDetails());
+            return false;
+        }
+
+        if (!answer.getResult()) {
+            s_logger.warn("Unable to update health checks data to router " + router.getHostName() + ", details : " + answer.getDetails());
+        }
+
+        return answer.getResult();
```

Comment:
we are allowed to ignore the results?

Reply:
This command pushes data to router and returns true/false. Any error/exception is logged already.